eras/allegra/impl/src/Cardano/Ledger/Allegra/UTxO.hs

```diff
@@ -19,6 +19,7 @@ import Cardano.Ledger.UTxO (EraUTxO (..))
 
 instance Crypto c => EraUTxO (AllegraEra c) where
   type ScriptsNeeded (AllegraEra c) = ShelleyScriptsNeeded (AllegraEra c)
+  type DepositInfo (AllegraEra c) = DPState c
```

Comment:
I couldn't figure out why we needed to make `DepositInfo` a new type family. could you explain?

Reply:
I know why :slightly_smiling_face: That is because `UTxO` module was recently moved into `cardano-ledger-core`, which does not know what `DPState` is. 

The consistent way to solve this would be to add ShelleyEraUTxO class into `Cardano.Ledger.Shelley.UTxO` and move the `getConsumedValue` function in that class:
```haskell
class EraUTxO era => ShelleyEraUTxO era where
  getConsumedValue :: HasField "_keyDeposit" pp Coin => pp -> DPState era -> UTxO era -> TxBody era -> Value era
```



The type family approach would make sense only if that type family `DepositInfo` had anyc chance of being set to a different type in different eras, but since it is always the same, type family approach isn't warranted

Reply:
Or even better move the whole `EraUTxO` class into `cardano-ledger-shelley` into `Cardano.Ledger.Shelley.UTxO`, this will solve this problem in the easiest possible fashion.

Reply:
This would require moving Certificates to Core, but it looks like DCert may become a type family, so that might complicate things. Of course we might split DPState into two parts.